test/spec/Radicle/Tests.hs

```diff
@@ -785,6 +785,22 @@ test_from_to_radicle =
         [ testForType (Proxy :: Proxy [Text]) ]
     , testGroup "Generic a => a"
         [ testForType (Proxy :: Proxy Foo) ]
+    , testGroup "Single constructor with selectors"
+        [ testForType (Proxy :: Proxy Bar) ]
+    , testGroup "Single constructor with no selectors"
+        [ testForType (Proxy :: Proxy Baz) ]
+    , testGroup "Constructors optional for single constructor datatypes"
+        [ let v = Vec $ fromList [ kw "bar"
```

Comment:
Would it not make more sense to have:

```
Bar "hello" 42 == { :constructor :bar :value { :bar1 "hello" :bar2 42 }
```

Reply:
That's departing a lot from what it is currently, and so would make it very hard to be backwards compatible. This test is simply stating that what we _currently_ have still works.

Reply:
Also I think we discussed this in the PR which added FromToRad in the first place; that `[:just 1]` and `:nothing` was a lot nicer that `{:constructor :just :value 1}` and `{:constructor :nothing}`.